src/screens/BagsStack/IndividualBag.tsx

```diff
@@ -1,10 +1,113 @@
 import React, { useEffect } from "react";
-import { Button, Text, View } from "react-native";
+import {
+  Button,
+  Dimensions,
+  FlatList,
+  StyleSheet,
+  Text,
+  View,
+} from "react-native";
+import DiscCard from "../../components/DiscCard";
 import Header from "../../components/Header";
 import { colors, typographyStyles, globalStyles } from "../../styles/Index";
 
 export default function IndividualBag(navStack: any) {
+  interface discData {
+    id: number;
+    brand: string;
+    discName: string;
+    speed: string;
+    glide: string;
+    turn: string;
+    fade: string;
+  }
+
   const bagData = navStack.route.params.bagData.item;
+  const discData = [
+    {
+      id: 1,
+      brand: "Innova",
+      discName: "Destroyer",
+      speed: "12",
+      glide: "5",
+      turn: "-1",
+      fade: "6",
+    },
+    {
+      id: 2,
+      brand: "Innova",
+      discName: "Destroyer",
+      speed: "12",
+      glide: "5",
+      turn: "-1",
+      fade: "6",
+    },
+    {
+      id: 3,
+      brand: "Innova",
+      discName: "Destroyer",
+      speed: "12",
+      glide: "5",
+      turn: "-1",
+      fade: "6",
+    },
+    {
+      id: 4,
+      brand: "Innova",
+      discName: "Destroyer",
+      speed: "12",
+      glide: "5",
+      turn: "-1",
+      fade: "6",
+    },
+    {
+      id: 5,
+      brand: "Innova",
+      discName: "Destroyer",
+      speed: "12",
+      glide: "5",
+      turn: "-1",
+      fade: "6",
+    },
+    {
+      id: 6,
+      brand: "Innova",
+      discName: "Destroyer",
+      speed: "12",
+      glide: "5",
+      turn: "-1",
+      fade: "6",
+    },
+    {
+      id: 7,
+      brand: "Innova",
+      discName: "Destroyer",
+      speed: "12",
+      glide: "5",
+      turn: "-1",
+      fade: "6",
+    },
+  ];
+
+  const formatData = (discData: any, numColumns: number) => {
```

Comment:
made for the use case of when user has more than multiples of 3 discs, for example: 4 discs, the 4th disc would go to the next row but take up full screen width. This function keeps the disc card width the same no matter how many discs are in someones bag.